pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/Common.java

```diff
@@ -501,6 +504,210 @@ private static String getRepeatedEqualsGeneration(String generatedCodeSoFar, Fie
+	 */
+	public static String getFieldsCompareToStatements(final List<Field> fields, String generatedCodeSoFar, File destinationSrcDir) {
+		for (Field f : fields) {
+			if (f.optionalValueType()) {
+				generatedCodeSoFar += getPrimitiveWrapperCompareToGeneration(f);
+			} else if (f.repeated()) {
+				throw new UnsupportedOperationException("Repeated fields are not supported in compareTo method");
+			} else {
+				if (f.type() == Field.FieldType.FIXED32 ||
+						f.type() == Field.FieldType.INT32 ||
+						f.type() == Field.FieldType.SFIXED32 ||
+						f.type() == Field.FieldType.SINT32) {
+					generatedCodeSoFar +=
+							"""
+							result = Integer.compare($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							    return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.UINT32) {
+						generatedCodeSoFar +=
+       						"""
+							result = Integer.compareUnsigned($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							    return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+
+				} else if (f.type() == Field.FieldType.FIXED64 ||
+						f.type() == Field.FieldType.INT64 ||
+						f.type() == Field.FieldType.SFIXED64 ||
+						f.type() == Field.FieldType.SINT64) {
+					generatedCodeSoFar +=
+							"""
+							result = Long.compare($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							    return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.UINT64) {
+					generatedCodeSoFar +=
+							"""
+							result = Long.compareUnsigned($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							    return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.BOOL) {
+					generatedCodeSoFar +=
+							"""
+							result = Boolean.compare($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							    return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.FLOAT) {
+					generatedCodeSoFar +=
+							"""
+							result = Float.compare($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							    return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.DOUBLE) {
+					generatedCodeSoFar +=
+							"""
+							result = Double.compare($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							     return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.STRING ||
+						f.type() == Field.FieldType.ENUM) {
+					generatedCodeSoFar += generateCompareToForObject(f);
+				} else if (f.type() == Field.FieldType.BYTES) {
+					generatedCodeSoFar +=
+							"""
+							result = $fieldName.compareTo(thatObj.$fieldName);
```

Comment:
Minor: in theory this can throw an NPE if one of the objects didn't initialize its bytes field for whatever reason. This can happen via `Model.newBuilder().bytesField(null).build()`.

I'd suggest to simply use the generateCompareToForObject() that you've introduced here.

Reply:
Yes, it's a good idea, thank you